homeassistant/components/camera/webrtc.py

```diff
@@ -81,20 +91,98 @@ class WebRTCClientConfiguration:
 
     configuration: RTCConfiguration = field(default_factory=RTCConfiguration)
     data_channel: str | None = None
+    get_candidates_upfront: bool = False
 
     def to_frontend_dict(self) -> dict[str, Any]:
         """Return a dict that can be used by the frontend."""
         data: dict[str, Any] = {
             "configuration": self.configuration.to_frontend_dict(),
+            "getCandidatesUpfront": self.get_candidates_upfront,
         }
         if self.data_channel is not None:
             data["dataChannel"] = self.data_channel
         return data
 
 
+@dataclass(frozen=True)
+class WebRTCSessionId:
+    """WebRTC session ID."""
+
+    session_id: str
+
+    def to_frontend_dict(self) -> dict[str, Any]:
```

Comment:
We should not have a notion of `to_frontend_dict`. We have `as_dict` as a pattern in all our Python objects that we want to move around (or rely on `dataclasses.asdict`). The frontend, or any other consumer of websocket APIs like our mobile apps, deal with our casing and converts it as needed.